crates/common/tedge_utils/src/fs.rs

```diff
@@ -23,26 +47,55 @@ pub fn atomically_write_file_sync(
     let mut file = std_fs::OpenOptions::new()
         .write(true)
         .create_new(true)
-        .open(&tempfile)?;
-
-    if let Err(err) = std::io::copy(&mut reader, &mut file) {
-        let _ = std_fs::remove_file(tempfile);
+        .open(&tempfile)
+        .context(
+            format!("could not create the temporary file {tempfile:?}"),
+            &dest,
+        )?;
+
+    if let Err(err) = std::io::copy(&mut reader, &mut file).context(
+        format!("could not copy the content to the temporary file {tempfile:?}",),
+        &dest,
+    ) {
+        let _ = std_fs::remove_file(&tempfile);
         return Err(err);
     }
```

Comment:
As this is done several time, I would try to wrap `tempfile` with a struct that remove the file on drop so we can write:

```suggestion
    std::io::copy(&mut reader, &mut file).context(
        format!("could not copy the content to the temporary file {tempfile:?}",),
        &dest,
    ) ?;
```

That said this would not work in the `async` case.